src/main/java/com/example/Restaurant_Management_System_REST_API/service/TableService.java

```diff
@@ -0,0 +1,103 @@
+package com.example.Restaurant_Management_System_REST_API.service;
+
+import com.example.Restaurant_Management_System_REST_API.DTO.TableDTO.TableDTO;
+import com.example.Restaurant_Management_System_REST_API.exception.NotFoundInDatabaseException;
+import com.example.Restaurant_Management_System_REST_API.model.entity.Reservation;
+import com.example.Restaurant_Management_System_REST_API.model.entity.Table;
+import com.example.Restaurant_Management_System_REST_API.repository.TableRepository;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class TableService {
+
+    private final TableRepository tableRepository;
+    private final ModelMapper modelMapper;
+
+    public TableDTO add(TableDTO tableDTO) {
+        Table table = modelMapper.map(tableDTO, Table.class);
+        tableRepository.save(table);
+
+        return modelMapper.map(table, TableDTO.class);
+    }
+
+    public List<TableDTO> findAll() {
+        List<TableDTO> tableList = new ArrayList<>();
+        tableRepository.findAll().forEach(table -> tableList.add(modelMapper.map(table, TableDTO.class)));
+
+        return tableList;
+    }
+
+    private Table findById(Long id) throws NotFoundInDatabaseException {
+        return tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+    }
+
+    public ResponseEntity<?> deleteById(Long id) throws NotFoundInDatabaseException {
+        Table tableToDelete = tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+
+        tableRepository.delete(tableToDelete);
+
+        return new ResponseEntity<>("Table with id " + tableToDelete.getId() + " has been deleted!", HttpStatus.OK);
+    }
+
+    void iterateAndSetTablesToReservationAndSave(Reservation reservation) throws NotFoundInDatabaseException {
+        for (Table table : reservation.getTables()) {
+            checkIfTableExist(table.getId());
+            List<Reservation> reservationList = new ArrayList<>();
+
+            if(table.getReservationList() != null)
+                reservationList = table.getReservationList();
```

Comment:
this is a brilliant idea, I changed the code accordingly but there is a problem during update(). In spite of initializing it to new ArrayList<>() it does not see it and states it is null value:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/db673c11-bf64-4a87-9e56-871ebc8477d7)

I have changed the fetchType to EAGER but it didnt help:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/30f77253-61f5-4ac2-8e31-33550b384137)


the method itself is much more clean:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/c8528e79-2249-4ad5-9ea0-6add6fa68c4d)

but unfortunately it doesnt work throwing NullPointerException here:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/e8d04e59-ec10-4297-bc0d-65b46700d6bd)

do you have any idea why it doesnt see it as empty ArrayList? maybe because of modelMapper?

do you have any suggestions how to overcome it?

Reply:
I have an idea, maybe I should obtain it using repository? now it is taken using get()

Reply:
I made this change but still the same problem. I dont know why it is behaving like that. Odd to me

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/fcab0161-89be-4c0d-8408-062a8880486b)

Reply:
I'll have to check it out. You can rollback not working changes to the state where it was working and create a separate branch for this improvement. If you have Trello or some other mean to keep track of tasks, it is a good idea to write it there :) For now you can leave the previous working version and fix it in another, smaller PR

Reply:
I found it!!!!!!!! I went for a walk thought about it, returned and found the problem and solution! I implemented it and it works, but now I need to go out so I will explain later what and why.
cheers!

Reply:
wow that was a lot of changes! the problem was that a line above the saving there was a private method which was setting list to null (the outcome of the previous owning side relation).
I made changes and I have committed them -> I added some new DTOS so that only necessary information is being shown for the client. Plus I had to remake some test methods.
All work fine now I checked.